src/core/variables.tf

```diff
@@ -894,3 +899,118 @@ variable "robots_indexed_paths" {
   type        = list(string)
   description = "List of cdn paths to allow robots index"
 }
+
+## Event hub
+variable "eventhub_sku_name" {
+  type        = string
+  description = "Defines which tier to use."
+  default     = "Basic"
+}
+
+variable "eventhub_capacity" {
+  type        = number
+  description = "Specifies the Capacity / Throughput Units for a Standard SKU namespace."
+  default     = null
+}
+
+variable "eventhub_maximum_throughput_units" {
+  type        = number
+  description = "Specifies the maximum number of throughput units when Auto Inflate is Enabled"
+  default     = null
+}
+
+variable "eventhub_auto_inflate_enabled" {
+  type        = bool
+  description = "Is Auto Inflate enabled for the EventHub Namespace?"
+  default     = false
+}
+
+variable "eventhub_zone_redundant" {
+  type        = bool
+  description = "Specifies if the EventHub Namespace should be Zone Redundant (created across Availability Zones)."
+  default     = false
+}
+
+variable "eventhub_ip_rules" {
+  description = "eventhub network rules"
+  type = list(object({
+    ip_mask = string
+    action  = string
+  }))
+  default = []
+}
+
+variable "eventhubs" {
+  description = "A list of event hub topics to add to namespace."
+  type = list(object({
+    name              = string
+    partitions        = number
+    message_retention = number
+    consumers         = list(string)
+    keys = list(object({
+      name   = string
+      listen = bool
+      send   = bool
+      manage = bool
+    }))
+  }))
+  default = [
+    {
+      name              = "SC-Contracts"
+      partitions        = 30
+      message_retention = 7
+      consumers         = []
+      keys = [
+        {
+          name   = "selfcare-wo"
+          listen = false
+          send   = true
+          manage = false
+        },
+        {
+          name   = "datalake"
+          listen = true
+          send   = false
+          manage = false
+        }
+      ]
+    }
+  ]
```

Comment:
Is possible to put it into the variables files, and not into defaults to be more simple to debug in the future?